pkg/policy/manager.go

```diff
@@ -531,10 +605,46 @@ func (m *manager) getPolicy(ctx context.Context, log *zap.Logger, id string) (*p
 	return &policy, nil
 }
 
+func (m *manager) validatePolicy(ctx context.Context, log *zap.Logger, policy *pb.PolicyEntity) error {
+	result, err := m.ValidatePolicy(ctx, &pb.ValidatePolicyRequest{Policy: policy.RegoContent})
+
+	message := &pb.ValidatePolicyResponse{
+		Policy: policy.RegoContent,
+	}
+
+	if result != nil && !result.Compile {
+		message.Errors = result.Errors
+	}
+
+	if err != nil || (result != nil && !result.Compile) {
+		s, _ := status.New(codes.InvalidArgument, "failed to compile the provided policy").WithDetails(message)
+		log.Error("failed to compile policy", zap.Error(s.Err()))
+
+		return s.Err()
+	}
+
+	return nil
+}
+
 func (m *manager) policiesAlias() string {
 	return m.indexManager.AliasName(policiesDocumentKind, "")
 }
 
+func (m *manager) incrementPolicyVersion(ctx context.Context, log *zap.Logger, policyId string) (uint32, error) {
+	updateResponse, err := m.esClient.Update(ctx, &esutil.UpdateRequest{
+		Index:      m.policiesAlias(),
+		DocumentId: policyCounterId(policyId),
+		Refresh:    m.esConfig.Refresh.String(),
+		Message:    &emptypb.Empty{},
+	})
```

Comment:
If we made that routing change, we'd have to add a `Routing` field to the `esutil.UpdateRequest` struct too